resources/assets/lib/profile-page/main.tsx

```diff
@@ -251,6 +284,67 @@ export default class Main extends React.Component<Props> {
     }
   };
 
+  @action
+  private readonly handleLazyLoadDone = (key: ProfileExtraPage, snapshot: Snapshot) => {
+    const element = this.extraPages[key].current;
+    if (element == null) {
+      return;
+    }
+
+    const diff = element.getBoundingClientRect().height - snapshot.bounds.height;
+
+    if (this.scrollTo.baseScrollY == null) {
+      this.scrollTo.baseScrollY = snapshot.scrollY;
+    } else {
+      // take the smaller value to ignore premature shifts
+      this.scrollTo.baseScrollY = Math.min(snapshot.scrollY, this.scrollTo.baseScrollY);
+    }
+
+    const marginTop = this.stickyHeaderOffset;
+    if (snapshot.bounds.bottom < marginTop) {
+      this.scrollTo.scrollBy += diff;
+      this.queueScroll();
+    } else if (snapshot.bounds.top < marginTop // new size goes off the top of visible area, happens at the bottom of page.
+        && snapshot.bounds.bottom > marginTop) {
+      this.scrollTo.scrollBy += diff;
+    }
+
+    // this is used to peg the page to the bottom when new sections load and a page near the bottom is supposed to be in focus
+    // otherwise the browser may shift the page up. The indexOf is to restrict it to pages that come after lazy loaded ones, otherwise
+    // non-lazy loaded sections at the end don't get pegged to the bottom.
+    if (this.pageJumpingTo != null
+      && this.pageJumpingTo !== 'main'
+      && this.displayedExtraPages.indexOf(this.pageJumpingTo) > this.displayedExtraPages.indexOf(key)) {
+      this.pageScrollIntoView(this.pageJumpingTo);
+    }
+
+    this.queueScroll();
+  };
+
+  private readonly handleLazyLoadGetSnapshot = (name: ProfileExtraPage) => {
+    const element = this.extraPages[name].current;
+    if (element == null) return;
+
+    return {
```

Comment:
probably want to check distance to bottom page so the page is positioned as expected when its initial scroll is the last section

...but that may cause weird behavior if the user scrolls around while loading 🤔 (always apply `diff` for `scrollBy` in that case instead of checking the `marginTop`?)

Reply:
I found that snapshots could happen before the final DOM tree was fully rendered which resulted in it thinking it was at the bottom when it actually wasn't 😔 
It's one of the reasons maintaining distance from the bottom got abandoned

Reply:
mark something if `pageJumptingTo` the last page?